### PR TITLE
Fix/docs improvements

### PR DESCRIPTION
**Pull request by JSabadin**
## Purpose

Improving documentation, adding missing components, resolving discrepancies between documentation and code, and adding new images.

### CODE REVIEW

File: luxonis_train/attached_modules/losses/README.md [hunk 1]
```diff
@@ -1,133 +1,276 @@
...
+| `ohem_ratio`      | `float`                          | `0.1`         | Fraction of elements (e.g., pixels) to keep based on highest loss values.                                                                  |
+| `ohem_threshold`  | `float`                          | `0.7`         | Threshold for hard example selection, computed as `-torch.log(torch.tensor(ohem_threshold))`; only losses above this value are considered. |
+| `weight`          | `list[float] \| None`            | `None`        | (Forwarded to `CrossEntropyLoss`) Manual rescaling weight for each class.                                                                  |
+| `reduction`       | `Literal["none", "mean", "sum"]` | `"mean"`      | (Forwarded to `CrossEntropyLoss`) Specifies the reduction method.                                                                          |
+| `label_smoothing` | `float` (0.0 to 1.0)             | `0.0`         | (Forwarded to `CrossEntropyLoss`) Amount of label smoothing to apply.                                                                      |
 
-## `EfficientKeypointBBoxLoss`
+______________________________________________________________________
```
Comment (klemen1999) [inline luxonis_train/attached_modules/losses/README.md:119]: Let's remove this explicit separators from the READMEs (if they are in others as well). If headings are correctly used this is handled automatically by markdown rendering.

File: luxonis_train/attached_modules/metrics/README.md
Comment (klemen1999) [inline luxonis_train/attached_modules/metrics/README.md:1]: IMO either we split it by tasks (like losses) or we atleast note what type of labels are required for it to work if they can't be nicely grouped together.
And let's include torchmetrics metrics into the "main list", no need for them to be in their own section (we can just say that we use torchmetrics directly for those)

Comment (klemen1999) [inline luxonis_train/attached_modules/metrics/README.md:1]: For OKS let's also mention the importance of picking correct sigmas (similarly as we do for mAP) and a short sentence on what this value means

File: luxonis_train/attached_modules/visualizers/README.md
Comment (klemen1999) [inline luxonis_train/attached_modules/visualizers/README.md:1]: Let's mention what are required labels to be present for each of the visualizers.

File: luxonis_train/callbacks/README.md
Comment (klemen1999) [inline luxonis_train/callbacks/README.md:1]: Let's write one sentence for PL callbacks as well on what they do so user gets a rough idea. And on top of PL callback section let's somehow mention that for specific params they should click the link to the original documentation

File: luxonis_train/nodes/README.md [hunk 2]
```diff
@@ -24,18 +24,26 @@ arbitrarily as long as the two nodes are compatible with each other. We've group
...
+  - [Classification Heads](#classification-heads)
+    - [`ClassificationHead`](#classificationhead)
+  - [Segmentation Heads](#segmentation-heads)
+    - [`SegmentationHead`](#segmentationhead)
+    - [`BiSeNetHead`](#bisenethead)
+    - [`DDRNetSegmentationHead`](#ddrnetsegmentationhead)
+  - [Bounding Box Detection Heads](#bounding-box-heads)
+    - [`EfficientBBoxHead`](#efficientbboxhead)
+    - [`PrecisionBBoxHead`](#precisionbboxhead)
+  - [Keypoint Detection Heads](#keypoint-detection-heads)
```
Comment (klemen1999) [inline luxonis_train/nodes/README.md:36]: Let's use "Instance Keypoint Detection Heads"

**Review by klemen1999 — COMMENTED** [review]
Let's also make sure that all code-blocks examples are up-to-date with newest version - 
didn't see any particular one that isn't but just as a general note

File: luxonis_train/attached_modules/metrics/README.md [hunk 3]
```diff
@@ -13,22 +17,44 @@ List of all the available metrics.
...
-- [F1Score](https://lightning.ai/docs/torchmetrics/stable/classification/f1_score.html)
-- [Precision](https://lightning.ai/docs/torchmetrics/stable/classification/precision.html)
-- [Recall](https://lightning.ai/docs/torchmetrics/stable/classification/recall.html)
+Works with tasks such as classification, segmentation, and anomaly detection.
+
+## JaccardIndex
+
+Jaccard Index (Intersection over Union) metric from the [`torchmetrics`](https://lightning.ai/docs/torchmetrics/stable/classification/jaccard_index.html) module.
+
+Works with tasks such as classification, segmentation, and anomaly detection.
```
Comment (klemen1999) [inline luxonis_train/attached_modules/metrics/README.md:30]: Let's make this "works with" part a bit more prominent, maybe make it bold?

File: luxonis_train/attached_modules/visualizers/README.md [hunk 4]
```diff
@@ -17,7 +17,7 @@ Visualizers are used to render the output of a node. They are used in the `visua
 
 ## `BBoxVisualizer`
 
-Visualizer for bounding boxes.
+Visualizer for bounding box detection task.
```
Comment (klemen1999) [inline luxonis_train/attached_modules/visualizers/README.md:20]: Same as metrics, let's make this bold so it is more highlighted

**Review by klemen1999 — APPROVED** [review]
